sync/class.jetpack-sync-module-full-sync.php

```diff
@@ -74,6 +74,11 @@ function start( $modules = null ) {
 			$modules['users'] = $user_module->get_initial_sync_user_config();
 		}
 
+		if( isset( $modules['posts'] ) && isset( $modules['posts']['meta'] ) ) {
```

Comment:
you might want to include `is_array( $modules['posts'] )` as part of this condition, as often `$modules['posts']` is a boolean.